ClusterClient.php

```diff
@@ -0,0 +1,107 @@
+ */
+class Credis_ClusterClient extends Credis_Client
+{
+    /**
+     * Name of the cluster as configured in redis.ini
+     * @var string|null
+     */
+    protected ?string $clusterName;
+
+    /**
+     * Hosts & ports of the cluster
+     * Eg: ['redis-node-1:6379', 'redis-node-2:6379', 'redis-node-3:6379', 'redis-node-4:6379']
+     * @var array|null
+     */
+    protected ?array $clusterSeeds;
+
+    /**
+     * Enable persistent connections
+     * @var bool
+     */
+    protected bool $persistentBool;
+
+
+    /**
+     * Creates a connection to the Redis server on host {@link $host} and port {@link $port}.
+     * $host may also be a path to a unix socket or a string in the form of tcp://[hostname]:[port] or unix://[path]
+     *
+     * @param string|null $clusterName Name of the cluster as configured in redis.ini
+     * @param array|null $clusterSeeds Hosts & ports of the cluster; eg: ['redis-node-1:6379', 'redis-node-2:6379']
+     * @param float|null $timeout Timeout period in seconds
+     * @param float|null $readTimeout Timeout period in seconds
+     * @param bool $persistentBool Flag to establish persistent connection
+     * @param string|null $password The authentication password of the Redis server
+     * @param string|null $username The authentication username of the Redis server
+     * @throws CredisException
+     */
+    public function __construct(?string $clusterName, ?array $clusterSeeds = [], $timeout = null, $readTimeout = null, $persistentBool = false, $password = null, $username = null, $tlsOptions = null)
+    {
+        $this->clusterName = $clusterName;
+        $this->clusterSeeds = $clusterSeeds;
+        $this->scheme = null;
+        $this->timeout = $timeout;
+        $this->readTimeout = $readTimeout;
+        $this->persistentBool = $persistentBool;
+        $this->standalone = false;
+        $this->authPassword = $password;
+        $this->authUsername = $username;
+        $this->selectedDb = 0; // Note: Clusters don't have db, but it's in superclass
+        if (is_array($tlsOptions) && count($tlsOptions) !== 0) {
+            $this->setTlsOptions($tlsOptions);
+        }
+        // PHP Redis extension support TLS/ACL AUTH since 5.3.0 // Note: Do we need this in Credis_ClusterClient?
+        $this->oldPhpRedis = (bool)version_compare(phpversion('redis'), '5.3.0', '<');
+    }
+
+    /**
+     * @return this
+     * @throws CredisException
+     */
+    public function connect()
+    {
+        if ($this->connected) {
+            return $this;
+        }
+        $this->close(true);
+        if (!$this->redis) {
+            $this->redis = new RedisCluster(
+                $this->clusterName,
+                $this->clusterSeeds,
+                $this->timeout,
+                $this->readTimeout,
+                $this->persistentBool, // Note:  This can't be $this->persistent, because it is string
+                ['user' => $this->authUsername, 'pass' => $this->authUsername],
```

Comment:
>'pass' => $this->authUsername

🤦

Reply:
resolved in https://github.com/JacobBrownAustin/credis/commit/1f467762a1f0bb1375fa391f0e793e98eb033f1b